qdev_wrappers/tests/dataset/test_doNd.py

```diff
@@ -0,0 +1,92 @@
+"""
+These are the basic black box tests for the doNd functions.
+"""
+
+from qdev_wrappers.dataset.doNd import do0d, do1d, do2d
+from qcodes.instrument.parameter import Parameter
+from qcodes import config, new_experiment
+from qcodes.utils import validators
+
+import pytest
+
+config.user.mainfolder = "output"  # set ouput folder for doNd's
+new_experiment("doNd-tests", sample_name="no sample")
+
+_param = Parameter('simple_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1)
+
+_paramComplex = Parameter('simple_complex_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1 + 1j,
+                   vals=validators.ComplexNumbers())
+
+_param_set = Parameter('simple_setter_parameter',
+                       set_cmd=None,
+                       get_cmd=None)
+
+
+def _param_func():
+    """
+    A private utility function.
+    """
+    _new_param = Parameter('modified_parameter',
+                           set_cmd= None,
+                           get_cmd= lambda: _param.get()*2)
+    assert _new_param.get() == 2
```

Comment:
We might be able to make this a little cleaner by separating clearly between tests and fixtures. This function is both (uses assert and returns something that gets used somewhere else)

Reply:
Making parameters fixtures sufficient for this distinction? Or you would like to see something here explicitly?

Reply:
Making them fixtures is a good thing, and then adding a simple test that uses that fixture and does the assert, instead of having the assert inside the fixture.